codice/ddf
# DDF-2008 Update RegistryReportActionProvider based on design refinement

#### What does this PR do?
This action provider was updated based on a further refinement of its purpose. While making these changes, test coverage was also improved. 

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@clockard @brendan-hofmann 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@shaundmorris

#### How should this be tested?
Build, Run Tests, Confirm action provider present on Search UI

This can be tested by following these steps:
1. Installing the registry app
2. Ingesting a registry metacard ingest -t rim:RegistryPackage /Users/PathToYourDDF/ddf/catalog/spatial/registry/registry-ebrim-transformer/src/test/resources/csw-full-registry-package.xml
3. Ingesting a second non-registry metacard
4. Open the search UI and do a wildcard search, the use the Details option to search for a metacard-tags contains registry search
5. Select the registry metacard returned by this second search and select the Action tab
6. Observe the URL generated when selecting Export Registry Metacard Information -> Display in new tab/window

#### Any background context you want to provide?
The action provider's canHandle method has been expanded to handle metacards as well as sources and configurations.

#### What are the relevant tickets?
DDF-2008 Original Action Provider Ticket
DDF-2142 Ticket that spurred revisiting the Action Provider design

#### Screenshots (if appropriate)
![ddf2008_1](https://cloud.githubusercontent.com/assets/11355332/15477688/de3f7ed6-20ca-11e6-8d9a-bad259b7f6db.png)
![ddf2008_2](https://cloud.githubusercontent.com/assets/11355332/15477694/e2d44170-20ca-11e6-8502-1d5f03aecd14.png)

#### Checklist:
